src/js/components/Header/Header.js

```diff
@@ -1,19 +1,83 @@
-import React from 'react';
-
+import React, { useContext, useEffect, useState } from 'react';
+import { ThemeContext } from 'styled-components';
+import { useForwardedRef } from '../../utils';
 import { Box } from '../Box';
 
-const Header = React.forwardRef(({ ...rest }, ref) => (
-  <Box
-    align="center"
-    as="header"
-    direction="row"
-    flex={false}
-    justify="between"
-    gap="medium"
-    {...rest}
-    ref={ref}
-  />
-));
+const Header = React.forwardRef(({ sticky, ...rest }, ref) => {
+  const theme = useContext(ThemeContext);
+  const containerRef = useForwardedRef(ref);
+  const [stickyStyles, setStickyStyles] = useState();
+
+  useEffect(() => {
+    // threshold inital value set to 0
+    const threshold = 0;
+    // number of pixels the document is currently scrolled
+    let lastScrollY = window.pageYOffset;
+    // run our event listener callback once in each requestAnimationFrame
+    let runEventListener = false;
+
+    const updateScrollDir = () => {
+      const scrollY = window.pageYOffset;
+      if (Math.abs(scrollY - lastScrollY) < threshold) {
+        runEventListener = false;
+        return;
+      }
+      if (scrollY === 0) {
+        runEventListener = false;
+        setStickyStyles();
```

Comment:
The end result is the same for the state value, but I think supplying undefined or null is more clear in the intent.

```suggestion
        setStickyStyles(undefined);
```